pelican/plugins/more_categories/more_categories.py

```diff
@@ -79,6 +78,18 @@ def create_categories(generator):
     )
     generator._update_context(['categories'])
 
+    # Add descendents and children
+    descendents = defaultdict(list)
+    children = defaultdict(list)
+    for category, articles in generator.categories:
+        for anc in category.ancestors[1:]:
```

Comment:
Sorry, this should be `category.ancestors[:-1]`

Reply:
Fixed =)

Also, for the "rearrange all commits into three commits" part. To be honest, I'm not sure how to do that .-. Could you do that? (Or tell me what I should be doing and I can do it)